sympy/core/power.py

```diff
@@ -1736,6 +1736,11 @@ def mul(d1, d2):
         if not d.is_positive:
             g = g.simplify()
             _, d = g.leadterm(x)
+            if not d.is_positive:
+                if g.is_zero:
+                    return f**e
```

Comment:
This block could come  immediately after simplification on line 1737. If g is zero, there is no need to compute the leadterm.